jepsen/src/jepsen/nemesis/time.clj

```diff
@@ -138,35 +138,50 @@
     (teardown! [_ test]
       (reset-time! test))))
 
-(defn reset-gen
-  "Randomized reset generator. Performs resets on random subsets of the tests'
+(defn reset-gen-select
+  "Randomized reset generator. Performs resets on a selected subsets of the test's
   nodes."
-  [test process]
-  {:type :info, :f :reset, :value (util/random-nonempty-subset (:nodes test))})
-
-(defn bump-gen
-  "Randomized clock bump generator. On random subsets of nodes, bumps the clock
-  from -262 to +262 seconds, exponentially distributed."
-  [test process]
-  {:type  :info
-   :f     :bump
-   :value (zipmap (util/random-nonempty-subset (:nodes test))
-                  (repeatedly (fn []
-                                (long (* (rand-nth [-1 1])
-                                         (Math/pow 2 (+ 2 (rand 16))))))))})
-
-(defn strobe-gen
-  "Randomized clock strobe generator. On random subsets of the test's nodes,
+  [select]
+  (fn [test process]
```

Comment:
You could also accomplish this by adding a single-arity form to bump-gen, strobe-gen, etc.

Reply:
I thought about doing this, but I figured a separate function would be more clear.